From 0cba7523c7460bc1851f91d6c77f1169fd65dacc Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Fri, 9 May 2025 17:35:43 +0800
Subject: [PATCH 097/115] mtk: mac80211: fix addba buffer size

When a HE station connect to an EHT MLD ap, the sta->sta.valid_links
would not be zero and ap would send ba request that HE station cannot
understand.
Use sta->sta.mlo to check if station is using mlo.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 net/mac80211/agg-tx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/agg-tx.c b/net/mac80211/agg-tx.c
index 9592f681..ab7c388a 100644
--- a/net/mac80211/agg-tx.c
+++ b/net/mac80211/agg-tx.c
@@ -473,7 +473,7 @@ static void ieee80211_send_addba_with_timeout(struct sta_info *sta,
 	sta->ampdu_mlme.addba_req_num[tid]++;
 	spin_unlock_bh(&sta->lock);
 
-	if (sta->sta.valid_links ||
+	if (sta->sta.mlo ||
 	    sta->sta.deflink.eht_cap.has_eht ||
 	    ieee80211_hw_check(&local->hw, STRICT)) {
 		buf_size = local->hw.max_tx_aggregation_subframes;
-- 
2.45.2

